src/AspNet.Security.OAuth.LinkedIn/LinkedInAuthenticationHandler.cs

```diff
@@ -58,6 +60,32 @@ public class LinkedInAuthenticationHandler : OAuthHandler<LinkedInAuthentication
             }
 
             var payload = JObject.Parse(await response.Content.ReadAsStringAsync());
+
+            if (Options.Fields.Contains(LinkedInAuthenticationConstants.EmailAddressField))
+            {
+                var emailAddressRequest = new HttpRequestMessage(HttpMethod.Get, Options.EmailAddressEndpoint);
+                emailAddressRequest.Headers.Add("x-li-format", "json");
+                emailAddressRequest.Headers.Authorization = new AuthenticationHeaderValue("Bearer", tokens.AccessToken);
+
+                var emailAddressResponse = await Backchannel.SendAsync(emailAddressRequest, Context.RequestAborted);
+                if (!emailAddressResponse.IsSuccessStatusCode)
+                {
+                    Logger.LogError("An error occurred while retrieving the email address: the remote server " +
+                                    "returned a {Status} response with the following payload: {Headers} {Body}.",
+                                    /* Status: */ emailAddressResponse.StatusCode,
+                                    /* Headers: */ emailAddressResponse.Headers.ToString(),
+                                    /* Body: */ await emailAddressResponse.Content.ReadAsStringAsync());
+
+                    throw new HttpRequestException("An error occurred while retrieving the email address.");
+                }
+
+                var emailAddressPayload = JObject.Parse(await emailAddressResponse.Content.ReadAsStringAsync());
+                var emailAddress = emailAddressPayload.SelectToken("elements[0].handle~.emailAddress").ToString();
```

Comment:
Could you do this without using `SelectToken(...)`? This won't exist in .NET Core 3.0 with the Newtonsoft.Json dependency removed (see #280), so will make it easier to port over to the new JSON APIs.

Reply:
I changed it to use Linq syntax, as other providers